com.unity.perception/Runtime/GroundTruth/LensDistortionCrossPipelinePass.cs

```diff
@@ -0,0 +1,181 @@
+        private Material m_LensDistortionMaterial;
+
+        private bool fInitialized = false;
+
+        public static readonly int _Distortion_Params1 = Shader.PropertyToID("_Distortion_Params1");
+        public static readonly int _Distortion_Params2 = Shader.PropertyToID("_Distortion_Params2");
+
+        public LensDistortion m_lensDistortion;
+        public float? lensDistortionOverride = null;        // Largely for testing, but could be useful otherwise
+
+        public RenderTexture m_TargetTexture;
+        private RenderTexture m_distortedTexture;
+
+        //private LensDistortion m_LensDistortion;
+
+        public LensDistortionCrossPipelinePass(Camera targetCamera, RenderTexture targetTexture)
+            : base(targetCamera)
+        {
+            m_TargetTexture = targetTexture;
+        }
+
+
+        public override void Setup()
+        {
+            base.Setup();
+            m_LensDistortionShader = Shader.Find(k_ShaderName);
+
+            var shaderVariantCollection = new ShaderVariantCollection();
+
+            if (shaderVariantCollection != null)
+                shaderVariantCollection.Add(new ShaderVariantCollection.ShaderVariant(m_LensDistortionShader, PassType.ScriptableRenderPipeline));
+
+            m_LensDistortionMaterial = new Material(m_LensDistortionShader);
+
+            if(shaderVariantCollection != null)
+                shaderVariantCollection.WarmUp();
+
+            // Set up a new texture
+            if (m_distortedTexture == null || m_distortedTexture.width != Screen.width || m_distortedTexture.height != Screen.height) {
+
+                if (m_distortedTexture != null)
+                    m_distortedTexture.Release();
+
+                m_distortedTexture = new RenderTexture(Screen.width, Screen.height, 0, RenderTextureFormat.ARGBFloat, RenderTextureReadWrite.Linear);
+                m_distortedTexture.enableRandomWrite = true;
+                m_distortedTexture.filterMode = FilterMode.Point;
+                m_distortedTexture.Create();
+            }
+
+            // Grab the lens distortion
+#if HDRP_PRESENT
+            // Grab the Lens Distortion from Perception Camera stack
+            var hdCamera = HDCamera.GetOrCreate(targetCamera);
+            var stack = hdCamera.volumeStack;
+            m_lensDistortion = stack.GetComponent<LensDistortion>();
+#elif URP_PRESENT
+            var stack = VolumeManager.instance.stack;
+            m_lensDistortion = stack.GetComponent<LensDistortion>();
+#endif
+
+            fInitialized = true;
+        }
+
+        protected override void ExecutePass(ScriptableRenderContext renderContext, CommandBuffer cmd, Camera camera, CullingResults cullingResult)
+        {
+            // Review: This has been removed since we may apply this pass to various stages, just want to confirm I'm
+            // not missing anything before removing it
+            //if (s_LastFrameExecuted == Time.frameCount)
```

Comment:
+1 on removing. I don't think there is anything wrong with running this multiple times in the same frame.